### PR TITLE
Added argsort Tensorflow frontend function

### PR DESCRIPTION
**Pull request by salvo96**

### CODE REVIEW

**Comment by salvo96** [comment]
#6964

**Comment by salvo96** [comment]
#7001

**Review by Ishticode — COMMENTED** [review]
Hi Thanks for the PR and the issue raised. I will wait for @CatB1t to take a look and will try to move forward with this asap.

**Comment by CatB1t** [comment]
Hello @salvo96 , please take a look at the reported issue, however, your frontend function should always return `int32` according to the rules specified by tensorflow here: https://www.tensorflow.org/api_docs/python/tf/argsort#returns . This will likely get your issue solved and the test will no longer fail. Thanks!

**Review by Ishticode — CHANGES_REQUESTED** [review]
Hi. Please check @CatB1t's comment above and along with this could you undo the change in `helpers/assertion.py`. 

Thanks a lot Mustafa for the help. Greatly appreciated🙂.

File: ivy/functional/frontends/tensorflow/general_functions.py [hunk 1]
```diff
@@ -8,6 +8,15 @@
 from ivy.functional.frontends.tensorflow.tensor import EagerTensor
 
 
+@to_ivy_arrays_and_back
+def argsort(values, axis=-1, direction='ASCENDING', stable=False, name=None):
+    if direction == 'DESCENDING':
+        descending = True
+    else:
+        descending = False
+    return ivy.argsort(values, axis=axis, descending=descending, stable=stable).astype("int32")
```
Comment (Ishticode) [inline ivy/functional/frontends/tensorflow/general_functions.py:17]: Can you just fix this minor lint by putting the args on newline for each?

**Review by Ishticode — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/tensorflow/general_functions.py (hunk 1 above)
Comment (salvo96) [reply ivy/functional/frontends/tensorflow/general_functions.py]: Now the file is fixed

**Review by Ishticode — APPROVED** [review]

**Comment by Ishticode** [comment]
Looks great @salvo96 Thanks for your contribution and patience. I wish you all the best. 🙂.

**Pull request merged**